### PR TITLE
Add ability to run TLS webmail without external NGINX.

### PR DESCRIPTION
**Pull request by briantopping**
PR provides ability for web endpoints to be TLS protected via nginx instance provided in `front` container. Also add ability to change DNS root domain for an installation from `cluster.local`.

### CODE REVIEW

File: mailu/README.md [hunk 1]
```diff
@@ -76,3 +79,13 @@ If neither `persistence.hostPath` nor `persistence.existingClaim` is set, a new
...
+
+## Ingress
+
+The default ingress is handled externally. In some situations, this is problematic, such as when webmail should be accessible
+ on the same address as the exposed ports. Kubernetes services cannot provide such capabilities without vendor-specific annotations.
+ 
+By setting `ingress.externalIngress` to false, the internal NGINX instance provided by `front` will configure TLS according to
+ `ingress.tlsFlavor` and redirect `http` scheme connections to `https`. 
+ 
+ CAUTION: This configuration exposes `/admin` to all clients with access to the web UI.
```
Comment (micw) [inline mailu/README.md:91]: Could you elaborate what this exactly means? Is this a security issue? Or is it the same as with the current ingress behaviour?

Comment (briantopping) [reply mailu/README.md:91]: It’s the same as the current behavior, which also doesn’t document the fact that admin is exposed. I’m happy to document it differently if it is confusing or alarming. Just thought users ought to know that they are exposing it.

**Comment by micw** [comment]
I would really like to align this PR with #26. Could you check if/how it works together?

**Comment by briantopping** [comment]
They should work together. Seems like that is focused on bigger goals with certmanager. I was just enabling existing functionality.

**Comment by briantopping** [comment]
I've cleaned up this branch so you can merge it. This is a smaller change that would be easier to integrate #26 on top of, IMO.

**Comment by micw** [comment]
I do agree, it's merged

**Comment by briantopping** [comment]
It looks like a good PR, it's just kind of dated at this point, especially after all the other recent merges. LMK if he's uninterested in bringing it up-to-date and maybe I can do it for him.

**Pull request merged**